src/main/scala/org/broadinstitute/dsde/firecloud/dataaccess/ElasticSearchDAOQuerySupport.scala

```diff
@@ -100,6 +99,15 @@ trait ElasticSearchDAOQuerySupport extends ElasticSearchDAOSupport {
       rp => query.must(researchPurposeSupport.researchPurposeFilters(rp, toLibraryAttributeName))
     }
 
+    val groupsQuery = boolQuery
+    // https://www.elastic.co/guide/en/elasticsearch/reference/2.4/query-dsl-exists-query.html
+    groupsQuery.should(boolQuery.mustNot(existsQuery(fieldDiscoverableByGroups)))
+    if (groups.nonEmpty) {
+      groupsQuery.should(termsQuery(fieldDiscoverableByGroups, groups.asJavaCollection))
+    }
+    groupsQuery.should(termsQuery(fieldWorkspaceId+".keyword" , workspaceIds.asJavaCollection))
```

Comment:
also - for future thought, *definitely* not for this PR ... I think the most likely case is that any given user will have access to multiple workspaces that aren't published into Library. Currently, the corpus of published workspaces changes relatively infrequently (on publish, on unpublish,  on delete, and potentially on reindex). As a brainstorm,  we could maintain an in-memory "whitelist"  of all the published workspace ids, which updates on any of the change  events, and  then prune the user's workspaceId list down to just those that actually exist in Library. This would keep the filter - and the request payload - smaller. Probably only an  issue for some power users that have thousands of workspaces that aren't published into Library.